tools/npm/npm-guide.md

```diff
@@ -0,0 +1,183 @@
+# Introduction
+
+### What is npm?
+
+**npm** stands for **Node Package Manager** and helps with installation and management of packages in Node.js by providing a useful command line interface.
+
+### What is a Node package?
+
+A Node package is a library of code that implements a functionality. Naturally, developers want to import and use it within a project instead of re writing it.
+
+### Where can i get the Node packages from?
+
+Users open source and publish their packages in [npm Registry](https://www.npmjs.com/search). Since these packages are open and free, developers usually check README, stats and stars of each package in [GitHub](https://www.github.com) in order to decide how useful is a library before using it.
```

Comment:
I change it to the following:

> Developers open source and publish their packages in [npm Registry](https://www.npmjs.com/search). 

> Since these packages are open and free, users are recommended to consult the Stats at the right of the web page of npm Registry for a package. For example see Stats of [lodash](https://www.npmjs.com/package/lodash) package. There is displayed the number of installments along with the GitHub link which provides the stars and README of each package in order to decide how useful is a library before using it.